### DISCUSSION TITLE
app start error with factory=True option enabled

### DISCUSSION THREAD

**Discussion by alexted** [Kludex/uvicorn#2056] [Potential Issue]
out of the blue, I started getting this error when launching my application:
```bash
/home/user/.cache/pypoetry/virtualenvs/my-app-O8YYcA2l-py3.11/bin/python /home/user/PycharmProjects/my-app/asgi.py 
INFO:     Will watch for changes in these directories: ['/home/user/PycharmProjects/my-app']
INFO:     Uvicorn running on http://127.0.0.1:5000 (Press CTRL+C to quit)
INFO:     Started reloader process [29497] using WatchFiles
ERROR:    Error loading ASGI app factory: Expected tuple, got str
```
my entry point file _asgi.py_:
```python
import uvicorn

if __name__ == "__main__":
    uvicorn.run(
        "src.app.factory:create_app",
        factory=True,
        host="127.0.0.1",
        port=5000,
        log_level="debug",
        reload=True,
        use_colors=True,
    )

```
As you have probably already realized, _create_app_ is a function that returns an instance of the fastapi application. 

If I comment out the _factory=True_ line, the error disappears and everything starts normally.
My question is - WTF?

**Comment by Kludex** [comment]
I cannot reproduce it. Can you create a minimal, reproducible example?

**Reply by alexted** [reply]
No problem. An example app can be found [here](https://github.com/alexted/my-app).
Just create an .env file in the root directory of the project with the following contents:
```bash
YUGABYTE_DSN=postgresql+asyncpg://postgres:@localhost:5433/core_service

IDP_URL=http://127.0.0.1:8080/
IDP_PUBLIC_KEY=qwer1234qwer1234qwer1234
IDP_CLIENT_SECRET=qwer1234qwer1234qwer1234
```
or set them as env vars.

**Reply by Kludex** [reply]
Please make it public.

**Reply by alexted** [reply]
done

**Reply by Kludex** [reply]
That's not an MRE.

**Reply by alexted** [reply]
Excuse me, how to understand what you wrote?

**Reply by Kludex** [reply]
https://stackoverflow.com/help/minimal-reproducible-example

**Reply by alexted** [reply]
Listen, you fucking bureaucrat, go to hell. What else do you need to do so your crooked hands can run your crappy library? You want me to chew it up and put it in your mouth? If you don't want to do anything, just say so, don't bother my head.

**Comment by Kludex** [comment]
I've reported, and banned the OP from interacting with the organization given the lack of etiquette.

I've also locked the discussion, since nothing productive can come up here.

Have a nice day! 🙏
